src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt

```diff
@@ -272,28 +275,30 @@ class JooqBurdenEstimateRepository(
 
         if (set.status == BurdenEstimateSetStatus.COMPLETE)
         {
-            throw OperationNotAllowedError("This burden estimate set has been marked as complete." +
+            throw InvalidOperationError("This burden estimate set has been marked as complete." +
                     " You must create a new set if you want to upload any new estimates.")
         }
 
-        if (type == BurdenEstimateSetTypeCode.STOCHASTIC)
-        {
-            stochasticBurdenEstimateWriter.addEstimatesToSet(setId, estimates, responsibilityInfo.disease)
-        }
-        else
-        {
-            centralBurdenEstimateWriter.addEstimatesToSet(setId, estimates, responsibilityInfo.disease)
-        }
-
+        getEstimateWriter(set).addEstimatesToSet(setId, estimates, responsibilityInfo.disease)
         changeBurdenEstimateStatus(setId, BurdenEstimateSetStatus.PARTIAL)
         updateCurrentBurdenEstimateSet(responsibilityInfo.id, setId, type)
     }
 
     override fun closeBurdenEstimateSet(setId: Int, groupId: String, touchstoneId: String, scenarioId: String)
     {
         // Check all the IDs match up
-        getSetFromResponsibilityPath(groupId, touchstoneId, scenarioId, setId)
-        changeBurdenEstimateStatus(setId, BurdenEstimateSetStatus.COMPLETE)
+        val (set, _) = getSetFromResponsibilityPath(groupId, touchstoneId, scenarioId, setId)
```

Comment:
Missed this on the last PR, but can we rename this function to `getSetWithResponsibilityInfo` so it's clearer what it returns?

Reply:
Done! https://github.com/vimc/montagu-api/pull/233/commits/44b8242fbdfc888deecea5a8eb101f0af929ff7e